mozregression/download_manager.py

```diff
@@ -234,7 +234,7 @@ def __init__(self, destdir, session=requests, persist_limit=None):
     def get_dest(self, fname):
         return os.path.join(self.destdir, fname)
 
-    def cancel(self, cancel_if=None):
+    def cancel(self, cancel_if=None, wait=False):
```

Comment:
Because this may block (in case the network is busy or something), and it is used in the gui, https://github.com/mozilla/mozregression/blob/91dc13a55125eb72db810f3eca4ededcc6b8f5ab/gui/mozregui/build_runner.py#L166

I don't want the GUI to be blocking. It works in the gui because we wait (after the mainwindow is closed) for thread completion, then we remove the folder if required, https://github.com/mozilla/mozregression/blob/master/gui/mozregui/main.py#L148-L150. this is synchronized and blocking - but done when no window is shown so the user usually don't notice it.

Maybe I can let this function as it was, and just use the new wait() call in the commandline main.py file - this can be a bit more readable.

@wlach thoughts or suggestions ?